vyper/ast/parse.py

```diff
@@ -82,6 +87,13 @@ def parse_to_ast_with_settings(
     module = vy_ast.get_node(py_ast)
     assert isinstance(module, vy_ast.Module)  # mypy hint
 
+    for k, v in loop_var_annotations.items():
+        loop_var_vy_ast = vy_ast.get_node(v["parsed_ast"])
+        loop_var_annotations[k]["vy_ast"] = loop_var_vy_ast
+        del loop_var_annotations[k]["parsed_ast"]
+
+    module._metadata["loop_var_annotations"] = loop_var_annotations
```

Comment:
i don't think we need to attach this into the metadata tbh. this should all be handled before analysis, and ideally before we even get to vyper AST construction.

Reply:
i think we can accomplish all this just with adding a `visit_For` into the visitor in this file